src/payment/PaymentForm.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm } from 'redux-form';
+import { FormattedMessage } from '@edx/frontend-i18n';
+
+import CardHolderInformation from './CardHolderInformation';
+
+class PaymentForm extends React.Component {
+  onSubmit(values) {
+    /* istanbul ignore next */
+    alert(JSON.stringify(values)); // eslint-disable-line
+  }
+
+  render() {
+    const { handleSubmit } = this.props;
+    return (
+      <form onSubmit={handleSubmit(this.onSubmit)}>
+        <CardHolderInformation />
+        <button type="submit" className="btn btn-primary btn-lg float-right">
+          <FormattedMessage
+            id="payment.form.submit.button.text"
+            defaultMessage="Place Order"
+            description="The label for the payment form submit button"
+          />
+        </button>
+      </form>
+    );
+  }
+}
+
+PaymentForm.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+};
+
+export default reduxForm({ form: 'payment' })(PaymentForm);
```

Comment:
This key `form` needs to match the key in `combineReducers` right? Might be worth making a note or exporting a constant like `storeName` as it's done for `paymentStoreName`